cpp/tests/wrappers/bools_test.cpp

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/wrappers/bool.hpp>
+
+#include <tests/utilities/cudf_gtest.hpp>
+#include <tests/utilities/type_lists.hpp>
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/type_list_utilities.hpp>
+
+struct Bool8Test
+  : public cudf::test::BaseFixture
+  , cudf::test::UniformRandomGenerator<uint8_t> {
+
+  Bool8Test()
+    : cudf::test::UniformRandomGenerator<uint8_t>{
+        std::numeric_limits<uint8_t>::lowest(), 
+        std::numeric_limits<uint8_t>::max()} {}
+
+  uint8_t rand(){ return this->generate(); }
+};
+
+template <typename T>
+struct Bool8CtorTest
+  : public cudf::test::BaseFixture
+  , cudf::test::UniformRandomGenerator<T> {
+
+  Bool8CtorTest()
+    : cudf::test::UniformRandomGenerator<T>{}{}
+
+  T rand(){ return this->generate(); }
+};
+
+/**
+ * @brief The number of test trials
+ */
+static constexpr int NUM_TRIALS{10000};
+
+using Bool8CastSourceTypes = cudf::test::Concat<cudf::test::NumericTypes, cudf::test::Types<bool>>;
+
+TYPED_TEST_CASE(Bool8CtorTest, Bool8CastSourceTypes);
+
+TEST_F(Bool8Test, TestBool8Constructor) {
+  for (int i = 0; i < NUM_TRIALS; ++i) {
+    uint8_t t{this->rand()};
+    cudf::experimental::bool8 const w{t};
+    EXPECT_EQ(w.operator uint8_t(), static_cast<uint8_t>(static_cast<bool>(t)));
```

Comment:
This will test if the cast is invoking the cast operator as it should be.
```suggestion
    EXPECT_EQ(static_cast<uint8_t>(w), static_cast<uint8_t>(static_cast<bool>(t)));
```

Reply:
Instead of explicitly invoking the conversion operator, let's do a cast, since that's how most people will actually use it (like in your hash specialization).

Reply:
Updated :+1: